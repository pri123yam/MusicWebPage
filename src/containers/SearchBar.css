.Topcomponent{
    background-color: pink;
    text-align: center;
    border-style: solid;
    border-color: palevioletred;
}
.InputBox{
    height: 4vh;
    width: 50%;
    font-size:100%; 
}
.buttonclass{
    height: 4vh;
    width: 10%;
}