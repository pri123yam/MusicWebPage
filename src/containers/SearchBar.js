import React,{Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {searchSongs} from '../actions/SearchSong'
import './SearchBar.css'
export class SearchBar extends Component{
    handleChange=(event)=>{
        var searchKey=event.target.value;
        this.props.searchSongs(searchKey);
    }
    render(){
        return(
            <div className="Topcomponent">
                <h2>MUSIC WEB PAGE</h2>
                <input className='InputBox'
                placeholder='type search text here' type="text"
                onChange={this.handleChange}
                />
            </div>
        )
    }
}

function matchDispatchToProps(dispatch){
    return bindActionCreators({searchSongs},dispatch);
}
export default connect(null,matchDispatchToProps)(SearchBar);