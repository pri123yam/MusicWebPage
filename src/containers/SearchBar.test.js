import React from 'react';
import {SearchBar} from './SearchBar';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({
    adapter: new Adapter()
});
describe('<SearchBar/>', () => {

    it('should renders <SearchBar/> component correctly', () => {
        const searchMock=jest.fn();
        const event = {
            target: {
                value: 'Search Key'
            }
        };
        const wrapper = Enzyme.shallow(
            <SearchBar
                searchSongs={searchMock}
            />
        );
        expect(wrapper.debug()).toMatchSnapshot();
        wrapper.find('.InputBox').simulate('change', event);
        expect(searchMock).toHaveBeenCalledWith('Search Key');
    });
});