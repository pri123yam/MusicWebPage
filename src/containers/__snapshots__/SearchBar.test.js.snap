// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the working of search bar should match the previous snapshot 1`] = `
"<div className=\\"Topcomponent\\">
  <h2>
    MUSIC WEB PAGE
  </h2>
  <input className=\\"InputBox\\" placeholder=\\"type search text here\\" type=\\"text\\" onChange={[Function]} />
</div>"
`;
